### ISSUE TITLE
[MNT] `binder` install option does not install optional dependencies used in notebooks

### ISSUE THREAD

**Issue by MatthewMiddlehurst** [aeon-toolkit/aeon#307]
The `binder` install option should install these so that all notebooks can be run. I'm not even sure if all notebooks run in their current state, so this would require a bit of a review.

**Comment by RohitP2005** [comment]
@MatthewMiddlehurst  hey i am interested in solving this issue, if its not yet assigned or worked upon

**Comment by baraline** [comment]
Hey @RohitP2005, I don't think anyone is working on this currently, as this is an old issue, we should first check its status.
A first step would be to try to run the notebooks on binders to see if you get any failures due to missing optional dependencies.

**Comment by baraline** [comment]
like this :
@aeon-actions-bot assign @RohitP2005

**Comment by RohitP2005** [comment]
Thanks @baraline

**Comment by MatthewMiddlehurst** [comment]
Is this still being worked on?

**Comment by RohitP2005** [comment]
Yes , i will probably finish it within two days, sorry for the delay

**Comment by RohitP2005** [comment]
The last time I tried, the build was successfully , but didn't run the notebooks.
It seems that the binder site is not working currently.
Can u guide me forwards

**Comment by MatthewMiddlehurst** [comment]
Seems the site is currently down.

**Comment by RohitP2005** [comment]
I guess, I will wait for the site to be back up running

**Comment by RohitP2005** [comment]
Hey @MatthewMiddlehurst, if you don't mind, could you help by building the Binder and sharing the URL? My builds are currently failing. Once it's up, I'll check the notebooks and update the Binder dependencies myself.

**Comment by MatthewMiddlehurst** [comment]
I don't think it works like that unfortunately. but I'm also not so familiar 🙂. I just tried the link and It works for me, just takes a bit if it the first time an image is created for the commit.

**Comment by RohitP2005** [comment]
hey @MatthewMiddlehurst  i tried runnig those notebooks and most of them ran with no issue, but several notebooks ran into this erro
```OMP: Info #276: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.```

am i missing smth here?

**Comment by MatthewMiddlehurst** [comment]
That should not be the case. I just checked `tsfresh.ipynb` and got the following as expected.

![Image](https://github.com/user-attachments/assets/3cfca779-53cb-4b57-bc7a-b20cce83f345)

**Comment by RohitP2005** [comment]
> That should not be the case. I just checked `tsfresh.ipynb` and got the following as expected.
> 
> ![Image](https://github.com/user-attachments/assets/3cfca779-53cb-4b57-bc7a-b20cce83f345)

I see, 
I will look into it once again and update you

**Comment by MatthewMiddlehurst** [comment]
Is this still in-progress?

**Comment by RohitP2005** [comment]
I find it difficult with the binder enviroment. I think you can assign this to someone else 
Sorry about it

**Comment by TUSHARDHOKRIYA** [comment]
@MatthewMiddlehurst I would like to work on this issue. May I proceed?

**Comment by baraline** [comment]
@TUSHARDHOKRIYA Yes, we have no active PR on this matter, you can work on it

**Comment by TUSHARDHOKRIYA** [comment]
@baraline Issue resolved-Binder Dependencies Fixed

 Summary
- Fixed missing packages in 'pyproject.toml' binder section
- All 62 notebooks now work in Binder
Full Report: https://github.com/TUSHARDHOKRIYA/repoo/blob/main/BINDER_REPORT.md

Key Changes:
- Added: matplotlib, seaborn, statsmodels, pyod, tensorflow
- Tested: Multiple Binder launches

Could you please review the changes and let me know if everything looks good?
Changes made in this repo https://github.com/TUSHARDHOKRIYA/repoo

**Comment by baraline** [comment]
Hi thanks ! 
Please create a pull request for us to be able to review your changes properly

**Comment by TUSHARDHOKRIYA** [comment]
@baraline please review this

**Comment by TUSHARDHOKRIYA** [comment]
@baraline Thank you for the review and merge, Im looking forward to working on and solving another issue.
